### PR TITLE
[#151203322]: feat: added default values from openapi specs

### PR DESCRIPTION
**Pull request by gunzip**
calling t.validate() sets default values

### CODE REVIEW

File: lib/utils/default.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+ */
+// tslint:disable:no-any
+export function withDefault<T extends t.Any>(
+  type: T,
+  defaultValue: t.TypeOf<T>
+): t.Type<any, t.TypeOf<T>> {
+  return new t.Type(
+    type.name,
+    (v: any): v is T => type.is(v),
+    (v: any, c: any) => type.validate(v !== undefined ? v : defaultValue, c),
```
Comment (cloudify) [inline lib/utils/default.ts]: `v !== undefined && v !== null` ?

File: lib/api/definitions/TimeToLive.ts [hunk 2]
```diff
@@ -15,4 +17,7 @@ import { WithinRangeNumber } from "../../utils/numbers";
 
 export type TimeToLive = WithinRangeNumber<3600, 31536000>;
 
-export const TimeToLive = WithinRangeNumber(3600, 31536000);
+export const TimeToLive = withDefault(
+  WithinRangeNumber(3600, 31536000),
+  3600 as any
```
Comment (cloudify) [inline lib/api/definitions/TimeToLive.ts]: this `as any` bothers me!

Comment (cloudify) [reply lib/api/definitions/TimeToLive.ts]: you can replace `any` with `TimeToLive`, makes more sense and avoids using `any`

**Comment by cloudify** [comment]
There's a little consistency problem, since default values get casted to a validated type, one could (by mistake) set a non valid value as default - to avoid this, the model generation script could validate the default values and trigger an error if the default value doesn't pass the type validation

File: lib/utils/default.ts (hunk 1 above)
Comment (gunzip) [reply lib/utils/default.ts]: probably it's better to not overwrite a property explicitely set to null, ie:

```
{
  foobar: null
}
```

Comment (gunzip) [reply lib/utils/default.ts]: probably it's better to not overwrite any property explicitely set to null

Comment (cloudify) [reply lib/utils/default.ts]: not sure about that, everywhere in the code there's no type for which `null` is a valid value, meaning that if `v === null`, validation fails

**Comment by gunzip** [comment]
I've added a module defined variable `const defaultValue = ...` 
so the program exists with failure when the module is required (ie. during tests)

File: api/templates/macros.njk [hunk 3]
```diff
@@ -0,0 +1,20 @@
+{% macro exportConst(defaultValue, type, typedef, quote = false) %}
+  {% if defaultValue !== undefined %}
+    {% set typeName = type + "X" %}
```
Comment (cloudify) [inline api/templates/macros.njk]: perhaps a better naming here would be nice, like `Base` instead of `X`

**Comment by gunzip** [comment]
done

**Pull request merged**